File: scripts/algorithms/alternative-implementations/tc/py/triangle_count.py
import sys
import time
from scipy.io import mmread
import numpy as np

filename = sys.argv[1]
G = mmread(filename)
start = time.time()
G_square = G @ G
nb_triangles = G_square.multiply(G).sum() / 3.0
fin = time.time()
print(fin - start)

File: scripts/algorithms/alternative-implementations/tc/jl/triangle_count.jl
using MatrixMarket
using SparseArrays

G = mmread(ARGS[1])
start = time_ns()
G_square = G * G
nb_triangles = sum(G_square .* G) / 3.0
fin = time_ns()
println((fin - start) * 1e-9)
